sklearn/utils/estimator_checks.py

```diff
@@ -1190,6 +1189,32 @@ def check_estimators_pickle(name, estimator_orig):
         unpickled_result = getattr(unpickled_estimator, method)(X)
         assert_allclose_dense_sparse(result[method], unpickled_result)
 
+    # check if input with nans pickles properly
+    if name in ALLOW_NAN:
+        # set random 10 elements to be np.nan
```

Comment:
I suspect we should just do this above, rather than repeating the picklability test.